packages/block-library/README.md

```diff
@@ -12,6 +12,62 @@ npm install @wordpress/block-library --save
 
 _This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for ES2015+ such as lower versions of IE then using [core-js](https://github.com/zloirock/core-js) or [@babel/polyfill](https://babeljs.io/docs/en/next/babel-polyfill) will add support for these methods. Learn more about it in [Babel docs](https://babeljs.io/docs/en/next/caveats)._
 
+## Building JavaScript for the browser
+
+If a `frontend.js` file is present in the block's directory, this file will be built along other assets, making it available to load from the browser.
+
+This enables us to, for instance, load this file when the block is present on the page in two ways:
```

Comment:
Instead of doing this manually, should we support a `frontend` entry in `block.json`. how did the discussions around that evolve? cc @gziolo

Reply:
That's the plan as discussed in other PR that was closed, so we miss the full context here.

Filters for processing `block.json` in `register_block_type_from_metadata` were added in WordPress 5.7, so it might be hard to process in the Gutenberg plugin. I guess we need to add it in WordPress core first, then add a shim after 5.8 is out to be able to use in the plugin with the assumption that we support two major versions of WP.

It's possible to add handling in `register_block_type` in the plugin with [`register_block_type_args`](https://developer.wordpress.org/reference/hooks/register_block_type_args/) hook.

Reply:
Ok makes sense, this PR is looking good then.

Reply:
In #29537 we talked about using `script`—  or a new `frontend` — attribute in block.json is something that we discussed, but at the moment files declared using this attributes are loaded everywhere. There's a [trac issue](https://core.trac.wordpress.org/ticket/50328) describing the issue in detail.

In the meantime, this PR intends to address only the build side of the feature leaving the discussion of how to load those built files for later — I added _manual_ usage instructions so that we can take advantage of this while the mentioned trac issue is pending.